crates/full-node/sov-sequencer/src/preferred/nonce_buffer_task.rs

```diff
@@ -691,10 +708,15 @@ fn err_invalid_nonce<S: Spec, Rt: Runtime<S>>(
     ))))
 }
 
+fn is_notready_error<S: Spec, Rt: Runtime<S>>(result: &TransactionReceiverResult<S, Rt>) -> bool {
+    matches!(result, Ok(Err(AcceptTxError::NotFullySynced(_))))
```

Comment:
Same here, let's match explicitly on each arm so that adding a new variant doesn't silently break any assumptions